src/frontend/js/tinymce-plugins/custom_link_input/plugin.js

```diff
@@ -0,0 +1,291 @@
+
+    async function getCompletions(query, id) {
+        const url = tinymceConfig.getAttribute("data-link-ajax-url");
+        const response = await fetch(url, {
+            method: "POST",
+            headers: {
+                "X-CSRFToken": getCsrfToken(),
+            },
+            body: JSON.stringify({
+                query_string: query
+            })
+        });
+        if (response.status != 200) {
+            return [];
+        }
+
+        const data = await response.json();
+        return [data.data, id];
+    }
+
+    // Ask if the user wants to append https if that prefix is likely missing
+    function checkUrlHttps(editor, url, handler) {
+        if (url.toLowerCase().startsWith("www.")) {
+            editor.windowManager.confirm(tinymceConfig.getAttribute("data-link-dialog-confirm_https-text"), function (s) {
+                if (s)
+                    handler("https://" + url);
+                else {
+                    handler(url);
+                }
+            });
+        } else {
+            handler(url);
+        }
+    }
+
+    function updateLink(editor, anchorElm, text, linkAttrs) {
+        anchorElm.textContent = text;
+
+        editor.dom.setAttribs(anchorElm, linkAttrs);
+        editor.selection.select(anchorElm);
+    };
+
+
+    tinymce.PluginManager.add('custom_link_input', function (editor, _url) {
+        function isAnchor(node) {
+            return node.nodeName.toLowerCase() === 'a' && node.href;
+        }
+        function getAnchor() {
+            var node = editor.selection.getNode();
+            return isAnchor(node) ? node : null;
+        };
+
+        const openDialog = function () {
+            let prev_search_text = "";
+            let prev_link_url = "";
+            let prev_selected_completion = "";
+
+            // Store the custom user url separately, so that it can be restored when required
+            let user_url = "";
+
+            // Stores the current request id, so that outdated requests get ignored
+            let ajax_request_id = 0;
+            let completion_items = [];
+
+            function updateDialog(api) {
+                let data = api.getData();
+
+                // Set the url either to the selected internal link or to the user link
+                if (prev_selected_completion != data.completions) {
+                    if (data.completions != "") {
+                        api.setData({ url: data.completions });
+                    } else {
+                        // restore the original user url
+                        api.setData({ url: user_url });
+                    }
+                }
+                prev_selected_completion = data.completions;
+
+                // Automatically update the text input to the url by default
+                data = api.getData();
+                if (data.url != data.text && prev_link_url == data.text) {
+                    api.setData({ text: data.url });
```

Comment:
I implemented it, but with a restriction that the link text does not get overwritten if it is not empty. I think otherwise this behavior could be annoying for the users, if they already entered a link text and then started to search for an internal link, since in this case their link text input would get overwritten.